### PR TITLE
feat: trezor wallet

### LINKED ISSUE #33: [Wallet]: Trezor Support

### PR DESCRIPTION
**Pull request by towanTG**
closes #33

### CODE REVIEW

File: apps/vite-playground/src/WalletPicker.tsx [hunk 1]
```diff
@@ -62,6 +70,9 @@ export const WalletPicker = ({ setWallet }: Props) => {
           return skClient.connectLedger(chains[0], derivationPath);
         }
 
+        case WalletOption.TREZOR:
+          await skClient.connectTrezor(chains, [...NetworkDerivationPath[chains[0]]]);
```
Comment (chillios-ts) [inline apps/vite-playground/src/WalletPicker.tsx]:
```suggestion
          await skClient.connectTrezor(chains, NetworkDerivationPath[chains[0]]);
```

File: apps/vite-playground/vite.config.ts [hunk 2]
```diff
@@ -30,6 +31,8 @@ export default defineConfig({
       '@thorswap-lib/types': resolve('../../packages/types/src'),
       '@thorswap-lib/walletconnect': resolve('../../packages/walletconnect/src'),
       '@thorswap-lib/web-extensions': resolve('../../packages/web-extensions/src'),
+      os: 'rollup-plugin-node-polyfills/polyfills/os',
+      stream: 'rollup-plugin-node-polyfills/polyfills/stream',
```
Comment (chillios-ts) [inline apps/vite-playground/vite.config.ts]: ✂️

File: apps/vite-playground/vite.config.ts [hunk 3]
```diff
@@ -42,4 +45,9 @@ export default defineConfig({
       define: { global: 'globalThis' },
     },
   },
+  build: {
+    rollupOptions: {
+      plugins: [nodePolyfills()],
+    },
+  },
```
Comment (chillios-ts) [inline apps/vite-playground/vite.config.ts:50]: ✂️

File: apps/vite-playground/package.json [hunk 4]
```diff
@@ -26,6 +27,7 @@
     "@types/react": "18.2.0",
     "@types/react-dom": "18.2.1",
     "@vitejs/plugin-react": "4.0.0",
+    "rollup-plugin-node-polyfills": "^0.2.1",
```
Comment (chillios-ts) [inline apps/vite-playground/package.json]: ✂️

File: packages/trezor/package.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    "@ethersproject/bignumber": "5.7.0",
+    "@ethersproject/providers": "5.7.0",
+    "@ethersproject/transactions": "5.7.0",
+    "@thorswap-lib/helpers": "workspace:*",
+    "@thorswap-lib/swapkit-entities": "workspace:*",
+    "@thorswap-lib/toolbox-evm": "workspace:*",
+    "@thorswap-lib/toolbox-utxo": "workspace:*",
+    "@thorswap-lib/types": "workspace:*",
+    "@trezor/connect-web": "9.0.7",
+    "bitcoinjs-lib": "5.2.0"
```
Comment (chillios-ts) [inline packages/trezor/package.json:14]:
```suggestion
    "@thorswap-lib/helpers": "workspace:*",
    "@thorswap-lib/swapkit-entities": "workspace:*",
    "@thorswap-lib/types": "workspace:*",
    "@trezor/connect-web": "9.0.7",
```

Those should be placed in devDependencies & peerDependencies, and be included on toolbox (bitcoinjs-lib and @ethers packages should be there already)

File: packages/trezor/src/signer/evm.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+    //@ts-ignore
+    const result = await TrezorConnect.ethereumSignTransaction({
+      path: `m/${derivationPathToString(derivationPath)}`,
+      transaction: baseTx,
+    });
+
+    if (!result.success) throw new Error(result.payload.error);
+    const { r, s, v } = result.payload;
+    debugger;
```
Comment (chillios-ts) [inline packages/trezor/src/signer/evm.ts:75]: ✂️

File: packages/trezor/src/signer/evm.ts [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+      to,
+      value: BigNumber.from(value).toHexString(),
+      gasLimit: BigNumber.from(gasLimit).toHexString(),
+      nonce: (nonce || (await provider.getTransactionCount(from, 'pending'))).toString(),
+      data,
+      maxFeePerGas: BigNumber.from(maxFeePerGas).toHexString(),
+      maxPriorityFeePerGas: BigNumber.from(maxPriorityFeePerGas).toHexString(),
+    };
+
+    //@ts-ignore
```
Comment (chillios-ts) [inline packages/trezor/src/signer/evm.ts:67]: it's better to use `@ts-expect-error` instead if possible

File: packages/trezor/src/trezor.ts [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+  Chain.Doge,
+  Chain.Ethereum,
+  Chain.BinanceSmartChain,
+  Chain.Litecoin,
+] as const;
+
+export const trezorTSParams = {
+  lazyLoad: true, // this param will prevent iframe injection until TrezorConnect.method will be called
+  manifest: {
+    email: 'towan@thorswap.finance',
```
Comment (chillios-ts) [inline packages/trezor/src/trezor.ts:29]: 😓 probably we should leave it blank? or add thorswap mail? 🤔

File: packages/trezor/src/trezor.ts [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+        if (!success)
+          throw new Error(
+            'Failed to get address: ' +
+              ((payload as { error: string; code?: string }).error || 'Unknown error'),
+          );
+
+        return payload.address;
+      };
+
+      const address = await getAddress();
```
Comment (chillios-ts) [inline packages/trezor/src/trezor.ts:169]: This feels unnecessary -> we should either extract functions is possible outside of it or just use directly form `TrezorConnect.getAddress` call

**Review by chillios-ts — CHANGES_REQUESTED** [review]

File: apps/vite-playground/vite.config.ts (hunk 2 above)
Comment (towanTG) [reply apps/vite-playground/vite.config.ts]: done

File: apps/vite-playground/package.json (hunk 4 above)
Comment (towanTG) [reply apps/vite-playground/package.json]: done

File: apps/vite-playground/vite.config.ts (hunk 3 above)
Comment (towanTG) [reply apps/vite-playground/vite.config.ts]: done

**Review by chillios-ts — APPROVED** [review]

**Pull request merged**